src/controllers/attachment/index.ts

```diff
@@ -91,28 +124,27 @@ export class attachment extends Controller {
   @Get('/{id}')
   @Response<NotFound>(404)
   @Response<BadRequest>(400)
+  @Produces('application/octet-stream')
+  @Produces('application/json')
   @SuccessResponse(200)
-  public async getById(
-    @Request() req: express.Request,
-    @Path() id: string,
-    @Header('return-type') type: 'json' | 'file'
-  ): Promise<unknown | Readable> {
+  public async getById(@Request() req: express.Request, @Path() id: string): Promise<unknown | Readable> {
     this.log.debug(`attempting to retrieve ${id} attachment`)
-    const { accept } = req.headers
     const [attachment] = await this.db.attachment().where({ id })
     if (!attachment) throw new NotFound('attachment')
     const { filename, binary_blob } = attachment
 
-    // log and default to octect-stream
-    if (type === 'json' || accept === 'application/json') {
-      this.setHeader('content-type', 'application/json')
-      try {
-        return JSON.parse(binary_blob)
-      } catch (err) {
-        this.log.warn('requested type failed, returning as octet')
+    const orderedAccept = parseAccept(req.headers.accept || '*/*')
+    if (filename === 'json') {
+      for (const mimeType of orderedAccept) {
```

Comment:
why do we need this logic? we are kinda already documented headers. I think we are doubling up here on checks or I'm missing something

Reply:
we're not doubling up at all. This is the same logic as from `inteli-api` as per the ticket requirements. This logic replaces the logic on `return-type` which is a duplicate headerin meaning of `Accept`. I may have missed some exception handling here but I think this is closer